crates/sui-proxy/src/peers.rs

```diff
@@ -193,25 +251,41 @@ impl SuiNodeProvider {
 
             loop {
                 interval.tick().await;
-
-                match Self::get_validators(rpc_url.to_owned()).await {
-                    Ok(summary) => {
-                        let peers = extract(summary);
-                        // maintain the tls acceptor set
-                        let mut allow = nodes.write().unwrap();
-                        allow.clear();
-                        allow.extend(peers);
-                        info!("{} peers managed to make it on the allow list", allow.len());
-                        JSON_RPC_STATE
-                            .with_label_values(&["update_peer_count", "success"])
-                            .inc_by(allow.len() as f64);
-                    }
+                let sui_validators_summary = match Self::get_validators(rpc_url.to_owned()).await {
+                    Ok(summary) => summary,
                     Err(error) => {
                         JSON_RPC_STATE
                             .with_label_values(&["update_peer_count", "failed"])
                             .inc();
-                        error!("unable to refresh peer list: {error}")
+                        error!("unable to refresh peer list: {error}");
+                        continue;
                     }
+                };
+                let bridge_validators_summary =
+                    match Self::get_bridge_validators(rpc_url.to_owned()).await {
+                        Ok(summary) => summary,
+                        Err(error) => {
+                            JSON_RPC_STATE
+                                .with_label_values(&["update_bridge_peer_count", "failed"])
+                                .inc();
+                            error!("unable to refresh sui bridge peer list: {error}");
+                            continue;
+                        }
+                    };
+                // iff both the `sui_validators` and `bridge_validators` lists are up to date, then clear and update the allow list
+                {
+                    let sui_validators = extract(sui_validators_summary);
+                    let bridge_validators = extract_bridge(bridge_validators_summary).await;
+
+                    let mut allow = nodes.write().unwrap();
+                    allow.clear();
+                    allow.extend(sui_validators);
+                    allow.extend(bridge_validators);
```

Comment:
For updating the sui validator set i do think it makes sense to maybe nuke the whole thing and replace it wholesale, but for the bridge node, since we have to do this awkward fetching of their pubkeys as an additional step (which i still don't like and think we should force this info to be encoded onchain, @longbowlu  can do we this instead?), I think we probably want to have a map from Bridge node -> Pubkey and we only remove a entry here if the bridge node is no longer in the committee and update if we're able to successfully reach that bridge node to fetch its key. if the node is still a part of the committee  but we happen to fail fetching its pubkey on one tick, we likely don't want to remove it from the set as it could result in losing some small window of data from that node.